lib/audio_service.dart

```diff
@@ -1099,7 +1101,7 @@ abstract class BackgroundAudioTask {
   /// Called once when this audio task is first started and ready to play
   /// audio, in response to [AudioService.start]. When the returned future
   /// completes, this task will be immediately terminated.
-  Future<void> onStart();
+  Future<void> onStart(Map params);
```

Comment:
Should this be optional, otherwise it will break the API?

Reply:
ok, I will make the change.

Reply:
I tried to figure out how to make this param optional, but I didint find a way. the only thing you need to do its put this param on your onStart function when you extend BackgroundAudioTask. If you don't pass this param when you call the function AudioService.start then an empty map will be passed.

Reply:
even if I make {Map params} optional I think it will break the api, because you will need to declare anyway.